### PR TITLE
Add constants for commonly used values.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
This is a consolidation effort, avoiding usage
of naked strings in codebase. Whenever possible
use constants which can be repurposed elsewhere.

## Motivation and Context
This also fixes `goconst ./...` reported issues.

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/auth-handler_test.go [hunk 1]
```diff
@@ -84,9 +84,9 @@ func TestGetRequestAuthType(t *testing.T) {
 			req: &http.Request{
 				URL: &url.URL{
 					Host:     "localhost:9000",
-					Scheme:   "http",
+					Scheme:   httpScheme,
 					Path:     "/",
-					RawQuery: "X-Amz-Credential=EXAMPLEINVALIDEXAMPL%2Fs3%2F20160314%2Fus-east-1",
+					RawQuery: "X-Amz-Credential=EXAMPLEINVALIDEXAMPL%2Fs3%2F2016, 20170314%2Fus-east-1",
```
Comment (donatello) [inline cmd/auth-handler_test.go]: What does the above line mean?

File: cmd/web-handlers_test.go [hunk 2]
```diff
@@ -747,7 +747,7 @@ func testUploadWebHandler(obj ObjectLayer, instanceType string, t TestErrHandler
 		}
 
 		req.Header.Set("Content-Length", strconv.Itoa(len(content)))
-		req.Header.Set("x-amz-date", "20160814T114029Z")
+		req.Header.Set("x-amz-date", "2016, 20170814T114029Z")
```
Comment (donatello) [inline cmd/web-handlers_test.go]: Looks like this is from a search-replace bug?

File: cmd/web-handlers_test.go [hunk 3]
```diff
@@ -1342,7 +1342,7 @@ func TestWebCheckAuthorization(t *testing.T) {
 	req, err = http.NewRequest("PUT", "/minio/upload/bucket/object", nil)
 	req.Header.Set("Authorization", "Bearer foo-authorization")
 	req.Header.Set("Content-Length", strconv.Itoa(len(content)))
-	req.Header.Set("x-amz-date", "20160814T114029Z")
+	req.Header.Set("x-amz-date", "2016, 20170814T114029Z")
```
Comment (donatello) [inline cmd/web-handlers_test.go]: search-replace bug.

File: cmd/web-handlers_test.go [hunk 4]
```diff
@@ -1426,7 +1426,7 @@ func TestWebObjectLayerNotReady(t *testing.T) {
 	req, err = http.NewRequest("PUT", "/minio/upload/bucket/object", nil)
 	req.Header.Set("Authorization", "Bearer "+authorization)
 	req.Header.Set("Content-Length", strconv.Itoa(len(content)))
-	req.Header.Set("x-amz-date", "20160814T114029Z")
+	req.Header.Set("x-amz-date", "2016, 20170814T114029Z")
```
Comment (donatello) [inline cmd/web-handlers_test.go]: search-replace bug.

File: cmd/web-handlers_test.go [hunk 5]
```diff
@@ -1543,7 +1543,7 @@ func TestWebObjectLayerFaultyDisks(t *testing.T) {
...
```
Comment (donatello) [inline cmd/web-handlers_test.go]: search-replace bug.

**Review by donatello — COMMENTED** [review]
Some search-replace typo/bugs

File: cmd/auth-handler_test.go (hunk 1 above)
Comment (donatello) [reply cmd/auth-handler_test.go]: Also looks like a search replace bug

Comment (harshavardhana) [reply cmd/auth-handler_test.go]: This is a bug

**Comment by harshavardhana** [comment]
Fixed all the comments @donatello @krisis

**Comment by harshavardhana** [comment]
I see no other comments.. Merging this in.

**Pull request merged**